### ISSUE TITLE
beta20: currentTrackDuration always 0

### ISSUE THREAD

**Issue by matijse** [spotify/ios-sdk#719]
I noticed in the new beta that `currentTrackDuration` of the `SPTAudioStreamingController` always stays 0. Also `currentTrackURI` is `nil` (while playing a track).

I've tried both `startWithClientId(clientId)` and `startWithClientId(clientId, audioController: SPTCoreAudioController(), allowCaching: true)` 

I use `playURI(url, startingWithIndex: 0, callback: ...)` to play a track. 

`currentPlaybackPosition` updates as expected.

**Comment by matijse** [comment]
This is also the case in the Demo/Example app. For example when I add the following code in the `playPause` function (line 58 in `ViewController.m`)

```
NSLog(@"currentPlaybackPosition: %f, currentTrackDuration: %f", self.player.currentPlaybackPosition, self.player.currentTrackDuration);
```

And I press Play/Pause after a few seconds, I get:

```
Simple Track Playback[3100:51875] currentPlaybackPosition: 9.113000, currentTrackDuration: 0.000000
```

**Comment by d-kunin** [comment]
`currentTrackDuration` well be removed in the next release, please use `didChangePlaybackState`

**Comment by matijse** [comment]
@d-kunin that is unfortunate to hear... We were using that to keep track of the progress, so we could do some thins just before the song ends. Also we show a countdown of the time left of a song rather than the current playback position. 

Is there any chance you can still include it in future releases? Or any other ways to get the time remaining?

Thanks in advance.

**Comment by d-kunin** [comment]
@matijse we might expose it, but I cant promise on that.

Could you use subscription-style instead of pulling style? `SPTPlaybackState.currentTrack.durationMs` contains duration info and `currentPlaybackPosition` is still functional.

**Comment by matijse** [comment]
@d-kunin Thanks. This is more or less the same I already used the `currentTrackDuration`. I used to observe that value, so now I can just observe the changes in `didChangePlaybackState`. 

The only difference is that I also used to ask the duration directly to the player at random points in time when I needed it. Now I save it manually in a local variable every time it changes, so I can still use it. That might still be a use-case where it would be usefull to leave the property `currentTrackDuration` on the player.
